types/react/test/index.ts

```diff
@@ -54,6 +54,72 @@ declare const container: Element;
 //
 // Top-Level API
 // --------------------------------------------------------------------------
+{
+    interface State {
+        inputValue: string;
+        seconds: number;
+    }
+    /**
+     * This is a "pre EcmaScript class property era" style of setting state within constructor.
+     * This occurs also if you need to provide som logic before mounting your component.
+     * To mitigate this error you need to provide state property definition upfront.
+     */
+    class SettingStateFromCtorComponent extends React.Component<Props, State, Snapshot> {
+        // uncomenting this fixes the error :)
+        // state: State;
+        constructor(props: Props) {
+            super(props);
+            // $ExpectError
+            this.state = {
+                inputValue: 'hello'
+            };
+        }
+        render() { return null; }
+    }
+
+    class BadlyInitializedState extends React.Component<Props, State, Snapshot> {
+        // $ExpectError
```

Comment:
I have here the same problem as before. getting dtslint error on TS 2.6 

![image](https://user-images.githubusercontent.com/1223799/41849345-e35fcaaa-7880-11e8-86cd-288536362b9a.png)

should I remove this ?